docs/DeveloperGuide.md

```diff
@@ -1125,34 +1125,73 @@ testers are expected to do more *exploratory* testing.
+1. Removing a contact's field while all contacts are being shown
+    1. Prerequisites: At least one contact in the shown list.
+
+    2. Test case: `remove 1 l/`<br>
+       Expected: First contact language field is removed. Language field in the person detail panel becomes empty.
+
+    3. Test case: `remove 1 l/ t/`<br>
+       Expected: First contact language and tag fields are removed. Language and tag field in the person detail card becomes empty.
+
+    4. Test case: `remove 1 t/friends`<br>
+       Expected: First contact tag is removed only if its tag name previously was `friends`. Otherwise, an error message is shown as the field does not exist in the SOCKet.
+
+    5. Test case: `remove 0 t/`<br>
+       Expected: No contact field is removed. An error message is shown as the given index is invalid. An error is logged in the console.
+
+    6. Test case: `remove 0`<br>
+       Expected: No contact field is removed. An error message is shown as the given syntax is invalid. An error is logged in the console.
 
 ### Deleting a contact
 
 1. Deleting a contact while all contacts are being shown
 
    1. Prerequisites: List all contacts using the `list` command. Multiple contacts in the list.
 
-   1. Test case: `delete 1`<br>
+   2. Test case: `delete 1`<br>
       Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
 
-   1. Test case: `delete 0`<br>
+   3. Test case: `delete 0`<br>
       Expected: No contact is deleted. Error details shown in the status message. Status bar remains the same.
 
-   1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
+   4. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
 
 ### Clearing all contacts or tags
 
-1. _{ more test cases …​ }_
+1. Clearing all contacts or contacts associated with the tags while all contacts are being shown
+
+   1. Test case: `clear`<br>
+      Expected: All contacts will be cleared.
+   
+   2. Test case: `clear t/`<br>
+      Expected: All contacts will be cleared.
+   
+   3. Test case: `clear t/friends`
+      Expected: All contacts with tag `friends` will be cleared.
+   
+   4. Test case: `clear t/enemy`
+      Expected: If there are no contact with tag `enemy`, no contact will be cleared. An error message is shown as no such tag was found in the SOCket.
 
 ### Sorting contacts
 
 1. _{ more test cases …​ }_
 
 ### Viewing a contact's detailed information
 
-1. _{ more test cases …​ }_
+1. Viewing a contact's detail while all the contacts are being shown
+
+   1. Prerequisites: At least one project in the list.
+
+   2. Test case: `view 1`<br>
+      Expected: First contact in the list is viewed. Detailed information of the contact is displayed in the person detail panel.
+   
+   3. Test case: `view 1 t/`<br>
```

Comment:
Perhaps "no contact is displayed in the details panel" as "viewed" doesn't really tell the reader anything